src/main/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImpl.java

```diff
@@ -0,0 +1,383 @@
+import uk.gov.companieshouse.api.accounts.model.filing.Link;
+import uk.gov.companieshouse.api.accounts.model.ixbrl.Account;
+import uk.gov.companieshouse.api.accounts.service.FilingService;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
+import uk.gov.companieshouse.api.accounts.util.ixbrl.accountsbuilder.AccountsBuilder;
+import uk.gov.companieshouse.api.accounts.util.ixbrl.ixbrlgenerator.DocumentGeneratorConnection;
+import uk.gov.companieshouse.api.accounts.util.ixbrl.ixbrlgenerator.IxbrlGenerator;
+import uk.gov.companieshouse.document.data.DocumentDescriptionHelper;
+import uk.gov.companieshouse.environment.EnvironmentReader;
+import uk.gov.companieshouse.environment.impl.EnvironmentReaderImpl;
+
+@Service
+public class FilingServiceImpl implements FilingService {
+
+    private static final String API_KEY_ENV_VAR = "CHS_API_KEY";
+    private static final String DISABLE_IXBRL_VALIDATION_ENV_VAR = "DISABLE_IXBRL_VALIDATION";
+    private static final String DOCUMENT_BUCKET_NAME_ENV_VAR = "DOCUMENT_BUCKET_NAME";
+    private static final String DOCUMENT_RENDER_SERVICE_END_POINT = "/document-render/store";
+    private static final String DOCUMENT_RENDER_SERVICE_HOST_ENV_VAR = "DOCUMENT_RENDER_SERVICE_HOST";
+    private static final String IXBRL_LOCATION = "s3://%s/%s/%s";
+    private static final String LINK_RELATIONSHIP = "accounts";
+    private static final String PERIOD_END_ON = "period_end_on";
+
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Autowired
+    private IxbrlGenerator ixbrlGenerator;
+    @Autowired
+    private AccountsBuilder accountsBuilder;
+
+    private EnvironmentReader environmentReader;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Filing generateAccountFiling(Transaction transaction, CompanyAccountEntity accountEntity)
+        throws IOException {
+        Filing filing = null;
+
+        if (isSmallFullAccount(accountEntity)) {
+            filing = generateAccountFiling(transaction,
+                getAccountType(AccountsType.SMALL_FULL_ACCOUNTS.getAccountType()));
+        }
+
+        return filing;
+    }
+
+    /**
+     * Gets the account type information.
+     *
+     * @param accountTypeName - accounts type name. e.g. small-full, abridged, etc.
+     * @return accounts type information.
+     */
+    private AccountsType getAccountType(String accountTypeName) {
+        return AccountsType.getAccountsType(accountTypeName);
+    }
+
+    /**
+     * Generate the filing for the account type passed in.
+     *
+     * @param transaction - transaction id
+     * @param accountsType - Account type information: account type, ixbrl's template name, account
+     * @return {@link Filing} - null or filing with the filing information (e.g. ixbrl location,
+     * accounts name, etc)
+     * @throws IOException -
+     */
+    private Filing generateAccountFiling(Transaction transaction, AccountsType accountsType)
```

Comment:
I would make `AccountType` a `Function<AccountType>` so you can use it with a lambda, this should get rid of the need for the `getAccountType` method.

Reply:
I will leave it as it is. but it is good shout